presto-main/src/main/java/com/facebook/presto/operator/scalar/TDigestFunctions.java

```diff
@@ -119,7 +119,7 @@ public static Block destructureTDigest(@SqlType("tdigest(double)") Slice input)
         BlockBuilder meansBuilder = DOUBLE.createBlockBuilder(null, tDigest.centroidCount());
         BlockBuilder weightsBuilder = INTEGER.createBlockBuilder(null, tDigest.centroidCount());
         for (Centroid centroid : tDigest.centroids()) {
-            int weight = centroid.getWeight();
+            int weight = (int) centroid.getWeight();
```

Comment:
@tdcmeehan - Is this just a nit fix? Doesn't seem inline with the other changes.

Reply:
It's to preserve compatibility with the existing function, which unfortunately defined the weight as an integer.  We need to follow up to make sure this function is also migrated, but it should be done separately as it's a relatively harmless bug.